### ISSUE TITLE
How to define reactive member in Model?

### ISSUE THREAD

**Issue by szekelyisz** [vuex-orm/vuex-orm#662]
One of my model classes has a plain boolean member (not defined as a field to stay out of the ORM model):

```typescript
@OrmModel('devices')
export class Device extends Model {
  // some fields
  public locateIndicator = false;
  // some logic that toggles locateIndicator internally every second
}
```

I have a corresponding component that displays information about the `device` entity (or `Device` class) that pulls the corresponding `device` from Vuex-ORM in the `created()` hook:

```typescript
import { Vue, Component, Prop } from 'vue-property-decorator';
import { Device, Status } from '../store/models/Device';

@Component
export default class DeviceListItem extends Vue {
  @Prop(String) readonly id!: string;

  device!: Device;

  created() {
    this.device = Device.findIn([this.id])[0];
    console.log('retrieved: ', this.device);
  }

  get locateIndicator() {
    return this.device.locateIndicator;
  }
}
```

In this component I want to use `Device.locateIndicator` as a reactive attribute (like `<q-btn :outline="locateIndicator>"`). But it looks like the attribute is not reactive. `<q-btn :outline="device.locateIndicator>"` doesn't work either.

What's the appropriate way to achieve this?

**Comment by szekelyisz** [comment]
So I found out that the entire `DeviceListItem.device` is not reactive. Even if I access it like `$store.state.entities.devices.data[id]`.

I wonder what's going on.

**Comment by cuebit** [comment]
Does `Device.findIn([this.id])[0]` have a value when the `created` hook is fired? I can't see why this would not work.

Alternatively, use a computed prop instead of setting data on the `created` hook.

**Comment by szekelyisz** [comment]
Yes, the correct data is retrieved from the store when the hook is run. I can use all fields from the instance, but nothing is reactive.

I also tried initializing it implicitly:

```typescript
export default class DeviceListItem extends Vue {
  @Prop(String) readonly id!: string;

  device: Device = Device.findIn([this.id])[0];
  // rest of the class
}
```

This yields the same results.

Using a computed property makes no difference, either.

```typescript
@Component
export default class DeviceListItem extends Vue {
  @Prop(String) readonly id!: string;

  // device: Device = Device.findIn([this.id])[0];

  get device() {
    return Device.findIn([this.id])[0];
  }
}
```

**Comment by szekelyisz** [comment]
If I use a computed prop such as

```typescript
  get device() {
    return Device.findIn([this.id])[0];
  }
```

then the component receives updates from Vuex-ORM, but not the non-field members of the `Device` class (such as the `locateIndicator` field in the very first listing) are not reactive.

On the other hand, if I use a member in the component's class like

```typescript
  device: Device = Device.findIn([this.id])[0];
```

then `locateIndicator` becomes reactive, but the model becomes static.

How can I have both? I wanna avoid storing `locateIndicator` in the model. It won't hurt, but it's not part of the logical structure I'm trying to map to Vuex-ORM.

**Comment by szekelyisz** [comment]
I got the following usual error on the console (unrelated to the problem):

> Property or method "name" is not defined on the instance but referenced during render. Make sure that this property is reactive, either in the data option, or for class-based components, by initializing the property. See: https://vuejs.org/v2/guide/reactivity.html#Declaring-Reactive-Properties.

This suggests that all reactive properties must be initialized when using class-based components, otherwise they won't be reactive.

So I'm wondering if this behavior comes from the interference between Vuex-ORM and class-based components.
